bindings/rust/standard/bench/src/s2n_tls.rs

```diff
@@ -306,9 +282,46 @@ impl TlsConnection for S2NConnection {
         while read_offset < data_len {
             match self.connection.poll_recv(data) {
                 Poll::Ready(bytes_read) => read_offset += bytes_read?,
-                Poll::Pending => return Err("unexpected pending".into()),
+                Poll::Pending => return Err("blocking".into()),
             }
         }
         Ok(())
     }
+    
+    fn send_shutdown(&mut self) {
+        // This will return `Ready` if the peer already closed their connection.
+        // It will return `Pending` if the peer hasn't responded yet. 
+        let _ = self.connection.poll_shutdown();
+    }
+
+    fn shutdown_completed(&mut self) -> bool {
+        assert!(matches!(self.connection.poll_shutdown(), Poll::Ready(_)));
```

Comment:
This should work locally, but would potentially fail on a real connection. Is the plan to only use this on local connections with no chance of the close_notify taking a bit longer than expected to arrive? If the close_notify hasn't arrived yet, you won't get "Ready" here.